### DISCUSSION TITLE
Auth.users confusion ( adding extra data, accessing this data )

### DISCUSSION THREAD

**Discussion by gplusdotgr** [supabase/supabase#11595] [Questions] [answered]
I'm a little (a lot) confused with auth.users . 
If auth.users isn't supposed to be access with the APIs, then I guess any business logic that requires accessing to the users information ( email, username ) needs a copy of the each user in a separate table (as described in the documention with the `profiles` table). 

Provided the above is accurate, what's the best approach to pass 'extra' metadata for the user in the signup call (`supabase.auth.signUp`)  ( like firstname, lastname etc) and then access this data? Do I add extra columns in `auth.users` and then using the trigger I copy them upon `signUp` ? or do I make a 2nd API call like
```js
await supabase
      .from('profiles')
      .update({ firstName: firstName, lastName: lastName, username: username })
      .match({ id: user.id });
```

**edit** I found another discussion, in which a modification to the trigger was recommended:
```plsql
begin
  insert into public.profiles (id, firstName, lastName, username)
  values (new.id, new.raw_user_meta_data->>'firstName', new.raw_user_meta_data->>'lastName', new.raw_user_meta_data->>'username');
  return new;
end;
```

```js
    const { data: user, error } = await supabaseClient.auth.signUp({
      email: email,
      password: password,
      options: {
        data: {
          firstName: firstName,
          lastName: lastName,
          username: username
        }
      }
    });
```

but unfortunately:
```
{"type":"success","status":200,"data":"[{\"status\":1,\"body\":2},500,{\"error\":3},\"Database error saving new user\"]"}
```

**Comment by gplusdotgr** [comment, marked as answer]
for future wanderers, postgres doesn't like uppercase chars in table column names.. I changed `firstName` to `first_name`

**Comment by agungsb** [comment]
I don't have any column with uppercase. But I am still getting this issue. Very confusing.

**Comment by seminull** [comment]
more for future wanderers

```
DECLARE
  full_name TEXT := new.raw_user_meta_data->>'name';
  first_name TEXT;
  last_name TEXT;

BEGIN
  first_name := SPLIT_PART(full_name, ' ', 1);
  last_name := SPLIT_PART(full_name, ' ', 2);

  insert into public.profiles(id, email, first_name, last_name, username, avatar_url)
values (new.id, new.email, first_name, last_name, new.raw_user_meta_data->>'username', new.raw_user_meta_data->>'avatar_url');

  return new;
END
```
